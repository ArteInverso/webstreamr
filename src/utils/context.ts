import { Request } from 'express';
import { Context } from '../types';
import { getDefaultConfig } from './config';

export const contextFromRequest = (req: Request): Context => {
  return {
    hostUrl: new URL(`${req.protocol}://${req.host}`),
    id: req.headers['X-Request-ID'] as string,
    ...(req.ip && { ip: req.ip }),
    config: req.params['config'] ? JSON.parse(req.params['config']) : getDefaultConfig(),
  };
};
