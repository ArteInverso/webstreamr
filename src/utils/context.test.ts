import { Request } from 'express';
import { contextFromRequest } from './context';

describe('contextFromRequest', () => {
  test('with config and ip', () => {
    const req = {
      protocol: 'https',
      host: 'localhost',
      headers: {
        'X-Request-ID': 'fake-id',
      },
      ip: '127.0.0.1',
      params: { config: '{"de":"on"}' },
    };

    expect(contextFromRequest(req as unknown as Request)).toMatchSnapshot();
  });

  test('without config', () => {
    const req = {
      protocol: 'https',
      host: 'localhost',
      headers: {
        'X-Request-ID': 'fake-id',
      },
      params: { },
    };

    expect(contextFromRequest(req as unknown as Request)).toMatchSnapshot();
  });
});
