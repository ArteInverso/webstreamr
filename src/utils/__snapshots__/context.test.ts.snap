// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`contextFromRequest with config and ip 1`] = `
{
  "config": {
    "de": "on",
  },
  "hostUrl": "https://localhost/",
  "id": "fake-id",
  "ip": "127.0.0.1",
}
`;

exports[`contextFromRequest without config 1`] = `
{
  "config": {
    "en": "on",
    "multi": "on",
  },
  "hostUrl": "https://localhost/",
  "id": "fake-id",
}
`;
